### PR TITLE
Remoted integration tests T1: active response send 

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|Closes https://github.com/wazuh/wazuh-qa/issues/1104|

## Description
This PR adds the remoted `test_active_command.py` test, to check if the manager correctly sends an active command to the agent.

## Logs example

- Remoted receives from active response socket `/var/ossec/queue/alerts/ar`:
```
2021/03/08 14:00:04 wazuh-remoted[57340] ar-forward.c:40 at AR_Forward(): DEBUG: Active response request received: (local_source) [] NRN 073 dummy-ar admin 1.1.1.1 1.1 44 (any-agent) any->/testing/testing.txt - -
```

- Agent receives from the manager:
```
2021/03/04 14:07:44 ossec-agentd[49945] receiver.c:92 at receive_msg(): DEBUG: Received message: '#!-execd dummy-ar admin 1.1.1.1 1.1 44 (any-agent) any->/testing/testing.txt - -
```

## Tests
active_response_tests

## Documentation

documentation_test_active_response

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/remote.py [hunk 1]
```diff
@@ -240,6 +240,16 @@ def get_protocols(all_protocols):
     return [valid_protocols, invalid_protocols]
 
 
+def callback_active_response_received():
+    msg = fr"DEBUG: Active response request received: (local_source) \[\] NRN 002 restart-wazuh0 admin 1.1.1.1 1.1 44 (agente-cualquiera) any->/carpeta/testing - -"
+    return monitoring.make_callback(pattern=msg, prefix=monitoring.REMOTED_DETECTOR_PREFIX)
+
+
+def callback_active_response_sent():
+    msg = fr"DEBUG: Active response sent: #!-execd restart-wazuh0 admin 1.1.1.1 1.1 44 (agente-cualquiera) any->/carpeta/testing - -"
```
Comment (Pac0x73706f74686f756e64) [inline deps/wazuh_testing/wazuh_testing/remote.py:249]: "agente-cualquiera"

Comment (Pac0x73706f74686f756e64) [reply deps/wazuh_testing/wazuh_testing/remote.py:249]: "carpeta/testing"

Please, be careful with this, the project language is English!

File: deps/wazuh_testing/wazuh_testing/tools/monitoring.py [hunk 2]
```diff
@@ -993,3 +994,32 @@ def wait_mtime(path, time_step=5, timeout=-1):
...
+        except TypeError:
+            agent_registered = -1
+
+    while agent_status != 'active':
+            time.sleep(1)
+            cursor.execute(f"select connection_status from agent where id={agent_id}")
+            agent_status = cursor.fetchone()[0]
+
+    cursor.close()
+    con.close()
```
Comment (Pac0x73706f74686f756e64) [inline deps/wazuh_testing/wazuh_testing/tools/monitoring.py:1025]: This is duplicated from https://github.com/wazuh/wazuh-qa/blob/1024-remote-active-response/deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py#L585

File: tests/integration/test_remoted/test_active_response/test_active_response.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# fixtures
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+
+def test_active_response_send(get_configuration, configure_environment, restart_remoted):
+    """Test if `wazuh-remoted` send correctly active response commands.
```
Comment (Pac0x73706f74686f756e64) [inline tests/integration/test_remoted/test_active_response/test_active_response.py:50]:
```suggestion
    """Test if `wazuh-remoted` sends active response commands.
```

File: tests/integration/test_remoted/test_active_response/test_active_response.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    Check if execd sent active response command to the remoted module. Also it ensure that agent receives active command
+    message from manager's remoted module.
```
Comment (Pac0x73706f74686f756e64) [inline tests/integration/test_remoted/test_active_response/test_active_response.py:53]:
```suggestion
    Check if execd sends active response command to remoted module and agent receives an active command message from the manager.
```

File: tests/integration/test_remoted/test_active_response/test_active_response.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    Raises:
+        AssertionError: if `wazuh-remoted` does not send correctly active response command or some of debug messages
```
Comment (Pac0x73706f74686f756e64) [inline tests/integration/test_remoted/test_active_response/test_active_response.py:56]:
```suggestion
        AssertionError: if `wazuh-remoted` does not send active response command or some of debug messages
```

File: tests/integration/test_remoted/test_active_response/test_active_response.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        is not created in the process
+    """
+    protocol_array = (get_configuration['metadata']['protocol']).split(",")
+    for protocol in protocol_array:
+        # Is necessary for UDP test set rcv_msg_limit to 1000
```
Comment (Pac0x73706f74686f756e64) [inline tests/integration/test_remoted/test_active_response/test_active_response.py:61]:
```suggestion
        # rcv_msg_limit of 1000 is necessary for UDP test 
```

**Review by Pac0x73706f74686f756e64 — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/tools/monitoring.py [hunk 7]
```diff
@@ -20,6 +20,7 @@
 import threading
 import time
 import yaml
+import sqlite3
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/tools/monitoring.py:23]: This import is unused.

File: docs/tests/integration/test_remoted/test_active_response/test_active_response.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+## General info
+
+|Tier | Number of tests | Time spent |
+|:--:|:--:|:--:|
+| 0 | 8 | 4m 16s |
+
+## Expected behavior
+
+- Fail if exed does not send to remoted the active response.
```
Comment (BraulioV) [inline docs/tests/integration/test_remoted/test_active_response/test_active_response.md:18]:
```suggestion
- Fail if wazuh-execd does not send to remoted the active response.
```

File: docs/tests/integration/test_remoted/test_active_response/test_active_response.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# Test active response
+
+## Overview
+These tests will check if active response command is sent correctly to the agent.
+
+## Objective
+
+The objective is to check that the manager correctly sent active response to the agent and that this one receives it.
```
Comment (BraulioV) [inline docs/tests/integration/test_remoted/test_active_response/test_active_response.md:8]: I would add more detail here about the test. In fact, the format of the active response messages, how they must appear in the agent, etc.

File: docs/tests/integration/test_remoted/test_active_response/test_active_response.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+## General info
+
+|Tier | Number of tests | Time spent |
+|:--:|:--:|:--:|
+| 0 | 8 | 4m 16s |
+
+## Expected behavior
+
+- Fail if exed does not send to remoted the active response.
+- Fail if remoted does not received the active response sent by execd.
```
Comment (BraulioV) [inline docs/tests/integration/test_remoted/test_active_response/test_active_response.md:19]:
```suggestion
- Fail if wazuh-remoted does not receive the active response sent by wazuh-execd.
```

File: docs/tests/integration/test_remoted/test_active_response/test_active_response.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+- Fail if agent does not receives active response message.
```
Comment (BraulioV) [inline docs/tests/integration/test_remoted/test_active_response/test_active_response.md:20]:
```suggestion
- Fail if the agent does not receive an active response message.
```

File: deps/wazuh_testing/wazuh_testing/remote.py [hunk 12]
```diff
@@ -563,3 +573,11 @@ def check_agent_received_message(message_queue, search_pattern, timeout=5, updat
...
     queue_monitor.start(timeout=timeout, callback=monitoring.make_callback(search_pattern, '.*'),
                         update_position=update_position)
+
+
+def escapee_callback_message(message):
+    escaped = message.translate(str.maketrans({"]":  r"\]",
+                                              "[":  r"\[",
+                                              "(":  r"\(",
+                                              ")":  r"\)"}))
+    return escaped
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/remote.py:583]: You don't need to do this. You can actually simplify it by using `re.escape(str)`.

**Review by BraulioV — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/remote.py (hunk 1 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/remote.py]: Done 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: deps/wazuh_testing/wazuh_testing/tools/monitoring.py (hunk 2 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/tools/monitoring.py]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: tests/integration/test_remoted/test_active_response/test_active_response.py (hunk 3 above)
Comment (Rebits) [reply tests/integration/test_remoted/test_active_response/test_active_response.py]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: tests/integration/test_remoted/test_active_response/test_active_response.py (hunk 4 above)
Comment (Rebits) [reply tests/integration/test_remoted/test_active_response/test_active_response.py]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: tests/integration/test_remoted/test_active_response/test_active_response.py (hunk 5 above)
Comment (Rebits) [reply tests/integration/test_remoted/test_active_response/test_active_response.py]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: tests/integration/test_remoted/test_active_response/test_active_response.py (hunk 6 above)
Comment (Rebits) [reply tests/integration/test_remoted/test_active_response/test_active_response.py]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: deps/wazuh_testing/wazuh_testing/tools/monitoring.py (hunk 7 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/tools/monitoring.py]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: docs/tests/integration/test_remoted/test_active_response/test_active_response.md (hunk 8 above)
Comment (Rebits) [reply docs/tests/integration/test_remoted/test_active_response/test_active_response.md]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: docs/tests/integration/test_remoted/test_active_response/test_active_response.md (hunk 10 above)
Comment (Rebits) [reply docs/tests/integration/test_remoted/test_active_response/test_active_response.md]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

Comment (Rebits) [reply docs/tests/integration/test_remoted/test_active_response/test_active_response.md]: Done in f24a9cb6c25af452389901b4e84a4c0590a4b051

File: docs/tests/integration/test_remoted/test_active_response/test_active_response.md (hunk 11 above)
Comment (Rebits) [reply docs/tests/integration/test_remoted/test_active_response/test_active_response.md]: Done in f24a9cb6c25af452389901b4e84a4c0590a4b051

File: deps/wazuh_testing/wazuh_testing/remote.py (hunk 12 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/remote.py]: Done in 20cadf88fd2d3f4777f24434f6ca2b8e95ff0f11

File: deps/wazuh_testing/wazuh_testing/remote.py [hunk 13]
```diff
@@ -576,12 +586,13 @@ def intercept_socket_data(data):
         control_service('start', daemon='wazuh-analysisd')
 
 
-def check_agent_received_message(message_queue, search_pattern, timeout=5, update_position=True, error_message=''):
+def check_agent_received_message(message_queue, search_pattern, escape=False, timeout=5, update_position=True, error_message=''):
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/remote.py:589]: New parameters must be added at the end to avoid collateral damages.

**Review by BraulioV — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/remote.py (hunk 13 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/remote.py]: Done in a3680eda20c727382a4c272a7e914c96689417d9

**Review by BraulioV — DISMISSED** [review]

**Review by Pac0x73706f74686f756e64 — APPROVED** [review]

File: deps/wazuh_testing/wazuh_testing/remote.py [hunk 14]
```diff
@@ -576,12 +586,14 @@ def intercept_socket_data(data):
...
 
-def check_agent_received_message(message_queue, search_pattern, timeout=5, update_position=True, error_message=''):
+def check_agent_received_message(message_queue, search_pattern, timeout=5, update_position=True, error_message='',
+                                 escape=False):
     """Allow to monitor the agent received messages to search a pattern regex.
 
     Args:
         message_queue (monitoring.Queue): Queue containing the messages received in the agent.
         search_pattern (str): Regex to search in agent received messages.
+        escape (bool): Flag to escape special characters in the pattern
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/remote.py:596]: Please put the documentation and parameters in order.

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/tools/__init__.py [hunk 15]
```diff
@@ -79,32 +79,51 @@ def get_service():
...
+
+ANALYSISD_ANALISIS_SOCKET_PATH= os.path.join(QUEUE_SOCKETS_PATH, 'analysis')
+ANALYSISD_QUEUE_SOCKET_PATH= os.path.join(QUEUE_SOCKETS_PATH, 'queue')
+AUTHD_SOCKET_PATH = os.path.join(QUEUE_SOCKETS_PATH, 'auth')
+EXECD_SOCKET_PATH = os.path.join(QUEUE_SOCKETS_PATH, 'com')
+LOGCOLLECTOR_SOCKET_PATH = os.path.join(QUEUE_SOCKETS_PATH, 'logcollector')
+MONITORD_SOCKET_PATH = os.path.join(QUEUE_SOCKETS_PATH, 'monitor')
+REMOTED_SOCKET_PATH = os.path.join(QUEUE_SOCKETS_PATH, 'request')
+SYSCHECKD_SOCKET_PATH = os.path.join(QUEUE_SOCKETS_PATH, 'syscheck')
+WAZUHDB_SOCKET_PATH = os.path.join(QUEUE_DB_PATH, 'wdb')
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/tools/__init__.py:95]:
```suggestion
WAZUH_DB_SOCKET_PATH = os.path.join(QUEUE_DB_PATH, 'wdb')
```

File: deps/wazuh_testing/wazuh_testing/tools/sockets.py [hunk 16]
```diff
@@ -37,3 +37,11 @@ def recv_response(_socket, size):
     sock.close()
 
     return response
+
+
+def send_ar_message(ar_command):
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/tools/sockets.py:42]:
```suggestion
def send_active_response_message(active_response_command):
```

File: deps/wazuh_testing/wazuh_testing/tools/sockets.py [hunk 17]
```diff
@@ -37,3 +37,11 @@ def recv_response(_socket, size):
...
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+
+    sock.connect(ACTIVE_RESPONSE_SOCKET_PATH)
+    sock.send(f"{ar_command}".encode())
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/tools/sockets.py:46]:
```suggestion
    sock.send(f"{active_response_command}".encode())
```

File: docs/tests/integration/test_remoted/test_active_response/test_active_response_ar_sending.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+```
+2021/03/04 14:07:44 ossec-agentd[49945] receiver.c:92 at receive_msg(): DEBUG: Received message: 
+'#!-execd dummy-ar admin 1.1.1.1 1.1 44 (any-agent) any->/testing/testing.txt - -
+```
+
+## General info
+
+|Tier | Number of tests | Time spent |
+|:--:|:--:|:--:|
+| 0 | 8 | 4m 16s |
```
Comment (jmv74211) [inline docs/tests/integration/test_remoted/test_active_response/test_active_response_ar_sending.md:41]: 4m 16s? Update the test time according to the new updates.

**Review by jmv74211 — DISMISSED** [review]

File: tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+                                    error_message='The expected event has not been found in ossec.log')
+
+            log_callback = remote.callback_active_response_sent(active_response_message)
+
+            wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                    error_message='The expected event has not been found in ossec.log')
+
+            remote.check_agent_received_message(agent.rcv_msg_queue, f'#!-execd dummy-ar admin 1.1.1.1 1.1 44 '
+                                                                     f'(any-agent) any->/testing/testing.txt - -',
+                                                                     escape=True)
```
Comment (BraulioV) [inline tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py:91]: Add this AR message to the `remote` module as a variable.

File: tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+    {'PROTOCOL': TCP, 'PORT': '1514'},
+    {'PROTOCOL': UDP, 'PORT': '1514'},
+    {'PROTOCOL': TCP_UDP, 'PORT': '1514'}
+
+]
+metadata = [
+    {'protocol': TCP, 'port': '1514'},
+    {'protocol': UDP, 'port': '1514'},
+    {'protocol': TCP_UDP, 'port': '1514'}
+]
```
Comment (BraulioV) [inline tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py:32]: Add here more tests changing the port a different one too

File: tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+
+import os
+import pytest
+import time
+
+import wazuh_testing.remote as remote
+import wazuh_testing.tools.agent_simulator as ag
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.sockets import send_active_response_message
+from wazuh_testing import UDP, TCP, TCP_UDP
```
Comment (BraulioV) [inline tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py:13]:
```suggestion
import wazuh_testing.remote as remote
import wazuh_testing.tools.agent_simulator as ag
from wazuh_testing import UDP, TCP, TCP_UDP
from wazuh_testing.tools.configuration import load_wazuh_configurations
from wazuh_testing.tools.sockets import send_active_response_message
```

**Review by BraulioV — CHANGES_REQUESTED** [review]

File: tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py (hunk 19 above)
Comment (Rebits) [reply tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py]: Done in 4e24efb4f8dcf8f8c05b75fe08b5820fb4f3305b,3cc7447ababd79cb7c49087dde0a2fbfe5cec5e1

File: tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py (hunk 20 above)
Comment (Rebits) [reply tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py]: Done in 3cc7447

File: tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py (hunk 21 above)
Comment (Rebits) [reply tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py]: Done in 3cc7447ababd79cb7c49087dde0a2fbfe5cec5e1

**Review by BraulioV — APPROVED** [review]

**Pull request merged**